chain/epoch_manager/src/shard_assignment.rs

```diff
@@ -136,15 +135,15 @@ fn assign_with_possible_repeats<T: HasStake + Eq, I: Iterator<Item = (usize, T)>
                 // assign a chunk producer right now.
                 let (least_validator_count, shard_stake, shard_id) =
                     shard_index.pop().expect("shard_index should never be empty");
-                if result[shard_pos].contains(&cp) {
+                if result[usize::try_from(shard_id).unwrap()].contains(&cp) {
                     buffer.push((least_validator_count, shard_stake, shard_id))
                 } else {
                     shard_index.push((
                         least_validator_count + 1,
                         shard_stake + cp.get_stake(),
                         shard_id,
                     ));
-                    result[shard_pos].push(cp);
+                    result[usize::try_from(shard_id).unwrap()].push(cp);
```

Comment:
let's add a unittest for this function, that would have caught this issue.

Reply:
Could I get an approval for this PR so we get the test failure fixed. We can worry about unit tests later.

Reply:
There actually is unrelated bug in `assign_with_possible_repeats` and it’ll take me more time to get another fix and unit tests so let’s get this merged. There already is NayDuck test that catches this failure and the test is currently failing.